docs/building-on-lisk/using-oracle-data/redstone-push.md

```diff
@@ -0,0 +1,215 @@
+  - address: `0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96`
+- [USDT/USD L2PriceFeedWithoutRounds](https://blockscout.lisk.com/address/0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD)
+  - address: `0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD`
+
+In this guide, we will develop a Consumer contract that will request the latest spot prices from all the above data feeds.
+
+:::note
+RedStone Push is only fully available on Lisk Mainnet, so please make sure to deploy your Consumer contract on Lisk Mainnet as well.
+
+In case you wish to deploy on Lisk Sepolia Testnet, check the [Tellor](./tellor.md) guide, which is available for both networks.
+:::
+
+## Import
+
+To use the RedStone data inside your contract, import the [AggregatorV3Interface](https://docs.chain.link/data-feeds/using-data-feeds#solidity) from Chainlink like shown in the example contract below.
+
+For every data feed you like to store, create a new constant with type `AggregatorV3Interface`.
+
+In the constructor, set the above defined constants to point to the respective data feeds:
+Use the `AggregatorV3Interface()` function and pass the adress of the respective data feed contract as parameter.
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
+
+contract RedStoneDataConsumer {
+    AggregatorV3Interface internal dataFeedETH;
+    AggregatorV3Interface internal dataFeedLSK;
+    AggregatorV3Interface internal dataFeedUSDT;
+
+    /**
+     * Network: Lisk
+     * Aggregator: ETH/USD
+     * Address: 0x6b7AB4213c77A671Fc7AEe8eB23C9961fDdaB3b2
+     */
+     /**
+     * Network: Lisk
+     * Aggregator: LSK/USD
+     * Address: 0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96
+     */
+      /**
+     * Network: Lisk
+     * Aggregator: USDT/USD
+     * Address: 0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD
+     */
+    constructor() {
+        dataFeedETH = AggregatorV3Interface(
+            0x6b7AB4213c77A671Fc7AEe8eB23C9961fDdaB3b2
+        );
+        dataFeedLSK = AggregatorV3Interface(
+            0xa1EbA9E63ed7BA328fE0778cFD67699F05378a96
+        );
+        dataFeedUSDT = AggregatorV3Interface(
+            0xd2176Dd57D1e200c0A8ec9e575A129b511DBD3AD
+        );
+    }
+}
+```
+
+## Reading data
+To read the data of the price feeds, we define the following functions in the contract:
+
+- `getRedStoneETHDataFeedLatestAnswer()`
+- `getRedStoneLSKDataFeedLatestAnswer()`
+- `getRedStoneUSDTDataFeedLatestAnswer()`
+
+Inside of the functions, call the [latestRoundData](https://docs.chain.link/data-feeds/api-reference#latestrounddata) on the respective data feeds to receive the latest spot prices for the respective token.
+
+The `latestRoundData()` function returns the following values:
+
+- `roundId`(uint80): The round ID.
+The "round" the answer was created in.
+Every time a price feed is updated, it adds +1 to the round ID.
```

Comment:
We're using data feed contracts without rounds, so round is always `1`:
Name of the contracts tells us this: `L2PriceFeedWithoutRounds`